repo/db/cases.go

```diff
@@ -254,7 +254,11 @@ func (c *CasesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sortByA
 		contract := new(pb.RicardianContract)
 		err := jsonpb.UnmarshalString(string(buyerContract), contract)
 		if err != nil {
-			jsonpb.UnmarshalString(string(vendorContract), contract)
+			err = jsonpb.UnmarshalString(string(vendorContract), contract)
+			if err != nil {
+				//return ret, 0, err
+				log.Error(err)
```

Comment:
Everything below relies on the successful unmarshaling of the contract. Any reason against continuing past this iteration which failed?